hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/insights/insights.tsx

```diff
@@ -89,8 +89,14 @@ export class Insights extends React.Component<Record<string, object>, IInsightsS
     // selected buckets value should be reset to all buckets
     let selectedBuckets = [allBucketsOption];
     // Update bucket options only if one volume is selected
-    if (selectedVolumes && selectedVolumes.length === 1) {
-      const selectedVolume = selectedVolumes[0].value;
+    if (((selectedVolumes && selectedVolumes.length != null) && (selectedVolumes.length === 2 && selectedVolumes[0].value === '*')) || (selectedVolumes.length === 1)){
```

Comment:
Is the `selectedVolumes.length != null` necessary? Later we are checking if the size is either 1 or 2.
```suggestion
    if ((selectedVolumes && (selectedVolumes.length === 2 && selectedVolumes[0].value === '*')) || (selectedVolumes.length === 1)){
```

Reply:
Yes Sometimes if length and selectedVolumes[0].value === '*' are both mandatory condition.

Reply:
I think the brackets wasn't at good place previously. So there is a `AND` after the `selectedVolumes`(like it was originally in the code) and after that inside the brackets there is an `OR`, where we need the length to be either 1 or 2. If we are checking that it either needs to be 1 or 2, why do we need the null check before that? Please check my suggestion in this comment.
```suggestion
    if (selectedVolumes && ((selectedVolumes.length === 2 && selectedVolumes[0].value === '*') || (selectedVolumes.length === 1))){
```

Reply:
Done Changes Suggested by Zita